ext/http/00_serve.ts

```diff
@@ -499,11 +500,15 @@ function mapToCallback(context, callback, onError) {
       innerRequest = new InnerRequest(req, context);
       const request = fromInnerRequest(innerRequest, "immutable");
       innerRequest.request = request;
-      response = await callback(
+      response = callback(
         request,
         new ServeHandlerInfo(innerRequest),
       );
 
+      if (typeof response === "object" && ReflectHas(response, "then")) {
```

Comment:
Should we add a null check here? Calling `Reflect.has(null, "then")` will throw, but not sure how strict we are in ensuring that the `callback` cannot return `null` prior to here.

```suggestion
      if (response !== null && typeof response === "object" && ReflectHas(response, "then")) {
```